### PR TITLE
Cleanup BitArray tests

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

**Comment by hughbe** [comment]
Test Innerloop OSX Debug Build and Test

File: src/System.Collections/tests/BitArray/BitArray_CtorTests.cs [hunk 1]
```diff
@@ -2,84 +2,84 @@
...
-            Assert.Equal(i, arraySize); //"Err_3! i should be equal to size"
-            Assert.Equal(bitArray.Length, arraySize); //"Err_4! ba1.Length should be equal to size"
+            BitArray bitArray = new BitArray(length);
+            Assert.Equal(length, bitArray.Length);
+            for (int i = 0; i < bitArray.Length; i++)
+            {
+                Assert.False(bitArray[i]);
+                Assert.False(bitArray.Get(i));
+            }
+            Ctor_BitArray(bitArray);
```
Comment (stephentoub) [inline src/System.Collections/tests/BitArray/BitArray_CtorTests.cs]: Clever

File: src/System.Collections/tests/BitArray/BitArray_OperatorsTests.cs [hunk 2]
```diff
@@ -2,242 +2,177 @@
...
-                Assert.True(ba4.Get(2)); //"Err_5! ba4.Get(2) should be true"
-
-                for (int i = 0x10000; i < ba2.Length; i++)
+                else
                 {
-                    if (i <= 0x10001)
-                        Assert.False(ba4.Get(i)); //"Err_6! ba4.Get(" + i + ") should be false"
-                    else
-                        Assert.True(ba4.Get(i)); //"Err_7! ba4.Get(" + i + ") should be true"
+                    Assert.False(bitArrayNot[i]);
```
Comment (stephentoub) [inline src/System.Collections/tests/BitArray/BitArray_OperatorsTests.cs]: Why are the if/else blocks the same?

Comment (hughbe) [inline src/System.Collections/tests/BitArray/BitArray_OperatorsTests.cs]: Good catch - fixed

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please

**Comment by hughbe** [comment]
@stephentoub I obviously ran a coverage report for System.Collections after cleaning up the tests. I expect a slight increase in coverage too. However, BitArray doesn't come up in the coverage report for some reason - it doesn't come up on the http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/code_coverage_windows/Code_Coverage_Report/ either

**Comment by stephentoub** [comment]
> However, BitArray doesn't come up in the coverage report for some reason

It doesn't come up in the coverage report because BitArray is coming from mscorlib; the implementation in corefx is only being compiled in for corert:
https://github.com/dotnet/corefx/blob/master/src/System.Collections/src/System.Collections.csproj#L49-L50
so you'll need to do a coverage run including mscorlib in order to see BitArray coverage.

@mellinoe, is this by design?  Is there a reason we can't use BitArray from corefx?

**Comment by ianhays** [comment]
LGTM.

>Is there a reason we can't use BitArray from corefx?

For coverage generation specifically, or for the build in general? I didn't think the latter was possible  since it's already exposed in mscorlib. The former isn't possible to my knowledge without the latter also being true i.e. to get coverage for BitArray using stock OpenCover we need to run the tests against the BitArray in CoreFX or we need to go through the trouble of running against a locally built mscorlib with accompanying pdb.

**Comment by stephentoub** [comment]
> for the build in general?

For build, i.e. code using BitArray would use one compiled into System.Collections.NonGeneric.dll rather than System.Collections.NonGeneric.dll type forwarding to the one in mscorlib.

> I didn't think the latter was possible since it's already exposed in mscorlib

Are there APIs exposed from mscorlib, other than BitArray itself, that use BitArray in a public signature?

**Comment by ianhays** [comment]
>Are there APIs exposed from mscorlib, other than BitArray itself, that use BitArray in a public signature?

Thanks for clarifying. I don't know of a good way to check this. Maybe I could remove BitArray from the exports list in mscorlib and then try building to see if I get errors from another class?

**Comment by stephentoub** [comment]
@hughbe, I checked the coverage of BitArray before and after your changes, and it looks like the branch coverage actually dropped.  Can you see why?

**Comment by hughbe** [comment]
@stephentoub thanks for running coverage reports - I'm getting some sort of infinite loop while running the tests, I think it may be related to the latest pdb/coverage changed in Buildtools. I'll try deleting and recloning corefx but this will take time - do you have a copy of the new coverage report on hand - perhaps you could link it on this PR if you have it (don't worry if you don't)

**Comment by stephentoub** [comment]
> do you have a copy of the new coverage report on hand

I don't, sorry.

**Comment by hughbe** [comment]
That's okay, I'll try to fix the coverage drop within the next day or so - sorr

**Comment by hughbe** [comment]
Test Innerloop Windows_NT Debug Build and Test

**Comment by hughbe** [comment]
Thanks @stephentoub for running a coverage report. I did some investigation and fixed the drop in coverage - I accidentally removed some test code that decreases then increases the size of the BitArray
At the same time, I added some more tests (for ICollection methods and properties, Ctor_IntArray and Ctor_ByteArray

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**